hw/ip/rv_dm/doc/dv/index.md

```diff
@@ -24,95 +15,131 @@ applicable. Once done, remove this comment before making a PR. -->
+This alert is wired from the bus integrity logic.
+
 The alert_agents provide the ability to drive and independently monitor alert handshakes via alert interfaces in RV_DM device.
 
 ###  JTAG Agent
-[Describe here or add link to its README]
 
-### UVC/agent 1
-[Describe here or add link to its README]
+The RVDM testbench instantiates an instance of [jtag_agent]({{<relref "hw/dv/sv/jtag_agent/doc">}}).
 
-### UVC/agent 2
-[Describe here or add link to its README]
+###  JTAG DMI Agent
 
-### UVM RAL Model
-The RV_DM RAL model is created with the [`ralgen`]({{< relref "hw/dv/tools/ralgen/README.md" >}}) FuseSoC generator script automatically when the simulation is at the build stage.
+The RV_DM testbench instantiates the sub-components of [jtag_dmi_agent]({{<relref "hw/dv/sv/jtag_dmi_agent/doc">}}) - the `jtag_dmi_monitor`, `sba_access_monitor` and the `jtag_dmi_reg_block`.
 
-It can be created manually by invoking [`regtool`]({{< relref "util/reggen/README.md" >}}):
+### UVM RAL Model
 
-### Reference models
-[Describe reference models in use if applicable, example: SHA256/HMAC]
+The RV_DM RAL model is created with the [`ralgen`]({{< relref "hw/dv/tools/ralgen/README.md" >}}) FuseSoC generator script automatically when the simulation is at the build stage.
+The input to the `reggen` tool is `hjson` file that contains the register descriptions.
+The specification captured in `hw/ip/rv_dm/data/rv_dm.hjson` only describes the registers on the main TL device interface.
+It does not contain the register space for the debug memory which is accessed via a separate TL device interface, since it comes from an external [PULP RISC-V debug](https://github.com/pulp-platform/riscv-dbg) repository.
+The `hw/ip/rv_dm/data/rv_dm_debug_mem.hjson` which is manually written and maintained by DV, describes this register space, and it is compatible with our `reggen` tool.
+The RAL model associated with it is thus, also generated similarly, with the `ralgen` FuseSoC generator script when simulations are run.
+The FuseSoC generator for creating these RAL models is added to `hw/ip/rv_dm/dv/env/rv_dm_env.core`.
+
+In addition, the RV_DM testbench instantiates the `jtag_agent`, which provides the JTAG DTM RAL model, describing the JTAG data transport module (DTM) registers.
+This RAL model is hand-written and comes with the `jtag_agent`.
+It is only accessible via the JTAG interface.
+The RV_DM environemnt also instantites the JTAG DMI RAL model, which describes the registers in the DMI register space (also accessible only via the JTAG interface).
+This register space is defined in the RISC-V debug specification.
+The PULP RISC-V debug implementation however only implements a subset of these registers.
+The RAL model is also limited to what is currently supported by the design.
+It is provided by the `jtag_dmi_agent`.
+
+All four of these RAL models are accessible (directly or indirectly) by the RV_DM envorinment configuration object.
```

Comment:
Just a thought, there is no need to address it in this PR.
Since you mentioned there are four RAL models, I wonder if it is easier to read if the above text list them out one by one. 
For example:
- RV_DM reg model:
- RV_DM debug memory model:
- JTAG DTM RAL model:
- JTAG DMI  RAL model:

Reply:
Good suggestion - updated. PTAL.